hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/execution/bulkinsert/RowCustomColumnsSortPartitioner.java

```diff
@@ -44,8 +44,8 @@ public RowCustomColumnsSortPartitioner(String[] columnNames) {
   @Override
   public Dataset<Row> repartitionRecords(Dataset<Row> records, int outputSparkPartitions) {
     final String[] sortColumns = this.sortColumnNames;
-    return records.coalesce(outputSparkPartitions)
-        .sortWithinPartitions(HoodieRecord.PARTITION_PATH_METADATA_FIELD, sortColumns);
+    return records.sort(HoodieRecord.PARTITION_PATH_METADATA_FIELD, sortColumns)
```

Comment:
Good catch, here maybe we can use `repartitionByRange` to gain the same behavior with the `RDDCustomColumnsSortPartitioner`

Reply:
@boneanxs `repartitionByRange` will not trigger sort, is there an API doing both sort and repartition with df? Now, I just implement it like `GlobalSortPartitionerWithRows` does

Reply:
I think `repartitionByRange` allows you to provide sort-order

Reply:
Yes, it partitions using rangepartitioner, but unordered within a single partition. Unless repartitionByRange then sortWithinPartitions